spec/models/order_spec.rb

```diff
@@ -63,6 +63,96 @@
     it{ is_expected.to have_db_column(:address_id).of_type(:integer)}
   end
 
+  describe 'priority rules' do
+    let(:at_6pm_today) { Time.now.in_time_zone.change(hour: 18) }
+    let(:at_6pm_yesterday) { at_6pm_today - 24.hours }
+    let(:after_6pm_today) { Time.now.in_time_zone.change(hour: 19) }
+    let(:after_6pm_yesterday) { after_6pm_today - 24.hours }
+    let(:before_6pm_today) { Time.now.in_time_zone.change(hour: 15) }
+    let(:before_6pm_yesterday) { before_6pm_today - 24.hours }
+
+    context 'A submitted order' do
+      it 'should be prioritised if it was submitted more than 24hours ago' do
+        old_order = create :order, state: "submitted", submitted_at: Time.now - 25.hours
+        order = create :order, state: "submitted", submitted_at: Time.now - 23.hours
+        expect(old_order.is_priority?).to eq(true)
+        expect(order.is_priority?).to eq(false)
+      end
+    end
+
+    context 'An order under review (aka processing)' do
+      
+      after { allow(Time).to receive(:now).and_call_original } # un-stub
+
+      context 'If we\'re past 6pm' do
+        before { allow(Time).to receive(:now).and_return(after_6pm_today) }
+
+        it 'should be prioritised if process was started before 6pm and hasn\'t finished' do
+          order_started_before_6 = create :order, state: "processing", processed_at: before_6pm_today
```

Comment:
Again just a suggestion you can add trait here in order factory.

create :order, state: "processing", processed_at: before_6pm_today

As this is repeated at multiple places.